### ISSUE TITLE
Show search bar for non-admin users

### ISSUE THREAD

**Issue by reggielamson** [Graylog2/graylog2-web-interface#620]
It would be nice if you could open up the following permission settings on the user edit pages in order for non-admin users to view the search bar and search page: 
public static final String SEARCHES_ABSOLUTE = "searches:absolute";
    public static final String SEARCHES_KEYWORD = "searches:keyword";
    public static final String SEARCHES_RELATIVE = "searches:relative";
    public static final String SAVEDSEARCHES_CREATE = "savedsearches:create";
    public static final String SAVEDSEARCHES_READ = "savedsearches:read";
    public static final String SAVEDSEARCHES_EDIT = "savedsearches:edit";

Reggie

**Comment by tramasoli** [comment]
+1

**Comment by JonMR** [comment]
Any movement on this one?

**Comment by rrodstein** [comment]
Greetings,

Could you please assist with a brief howto? FYI, We are looking for a way for our read only users to be able to search. 

Thank you,

"To put it simple, you just need to update the non-admin user permissions through the USER API. To achieve that just call the PUT /users/{username}/permissions with the following JSON":

{
"permissions": [
"searches:absolute",
"searches:keyword",
"searches:relative",
"messagecount:read",
"messages:read",
"throughput:read",
"savedsearches:create",
"savedsearches:read",
"users:passwordchange:monitoria",
"fieldnames:read",
"inputs:read",
"jvmstats:read",
"metrics:read",
"buffers:read",
"savedsearches:edit",
"users:edit:monitoria",
"indexercluster:read",
"system:read" 
]
}

**Comment by bernd** [comment]
Will be fixed once Graylog2/graylog2-server#1321 is in. We are currently working on this.

**Comment by avongluck-r1soft** [comment]
WHOO.  Figured it out using the new role system...

```
curl -v -XPOST -H 'Content-Type: application/json' 'http://ADMIN:PASSWORD@127.0.0.1:12900/roles' -d '{"read_only": false,"permissions": ["searches:absolute", "searches:keyword", "searches:relative"],"name": "Global Search","description": "Permission to perform global search"}'
```

That creates a new role that you can perform searches with when assigned.  YMMV... don't blame me if your graylog environment explodes.

**Comment by avongluck-r1soft** [comment]
(tested in 1.3.2)

**Comment by kroepke** [comment]
That will work, but most likely you will be missing necessary permissions
to show certain things in the message details (nodes, inputs, streams,
creating extractors etc). That's the primary reason we haven't enabled this
role yet, because it can lead to errors. But if this works for you so far,
go ahead! :)

**Comment by zahnd** [comment]
I tested this in 2.0.3 and it won't work...
![screen shot 2016-06-22 at 08 51 26](https://cloud.githubusercontent.com/assets/3197552/16257138/bf768e4c-3856-11e6-8b8e-4daecc004330.png)

Even if the user has read access to all streams.

**Comment by edmundoa** [comment]
@zahnd what did you test exactly? This issue is really old and quite a few things have been discussed.

If you have any questions about this, or how to do something, I would recommend posting it into our [public mailing list](https://groups.google.com/forum/#!forum/graylog2) or asking in the [#graylog channel on freenode IRC](https://webchat.freenode.net/?channels=%23graylog).

Thank you!

**Comment by taupehat** [comment]
@edmundoa I'm seeing the same thing as @zahnd did: recursive 403 errors cycling through when logging on as a user granted the newly-created "Global Search" role as described by @avongluck-r1soft 
Also on 2.0.3 here.

Should mention, the IRC channel is pretty quiet.

**Comment by edmundoa** [comment]
@taupehat Thank you for your feedback, I will try to reproduce the issue with that role and open a new issue if I can reproduce it.

Regarding the IRC channel, it is a community resource that depends on people being connected in that exact moment. We are usually around, but not always. If you need more reliable answers, please use our [professional support](https://www.graylog.org/professional-support) or [mailing list](https://groups.google.com/forum/?hl=en#!forum/graylog2).

**Comment by edmundoa** [comment]
I was taking a quick look at the role that @avongluck-r1soft kindly shared, and I had to add the `streams:read` permission to make it work in 2.0.3 / 2.1.0-alpha:

```
"permissions": [
    "searches:absolute",
    "searches:keyword",
    "searches:relative",
    "streams:read"
]
```

There may be other missing permissions, so you may need to update the role if you still get some 403s.

**Comment by AndresSomadossi** [comment]
Hi! Im using graylog-2.1.2. When i try to create a new role with:

"permissions": [
    "searches:absolute",
    "searches:keyword",
    "searches:relative",
    "streams:read"
]

The users with this role keep in loop and not work. Anyone know how fix it? Thank you!

**Comment by iamnut** [comment]
in case for @edmundoa must add `"streams:read"` permisstion into role

**Comment by thomagos** [comment]
Sad but true - this one https://github.com/graylog-labs/graylog2-web-interface/issues/620#issuecomment-165890069 is still valid for 2.3.1

Keep in mind that it's `127.0.0.1:9000/api/roles` for new installations.

**Comment by igor-pinchuk** [comment]
https://github.com/graylog-labs/graylog2-web-interface/issues/620#issuecomment-165890069 valid for 2.4.3 as well...

curl -v -XPOST -H 'Content-Type: application/json' 'http://ADMIN:PASSWORD@127.0.0.1:9000/api/roles' -d '{"read_only": false,"permissions": ["searches:absolute", "searches:keyword", "searches:relative"],"name": "Global Search","description": "Permission to perform global search"}'

**Comment by atc0005** [comment]
> @bernd: Will be fixed once Graylog2/graylog2-server#1321 is in. We are currently working on this.

> @tschwaerzl Sad but true - this one #620 (comment) is still valid for 2.3.1
> 
> Keep in mind that it's 127.0.0.1:9000/api/roles for new installations.

Should this lack of functionality be considered a regression or was this support never added? 

It appears that if you opt to use unique index sets for streams _and_ you opt to remove the messages for other streams from the default stream (presumably in an effort to prevent duplication of content between two sets), then you lose the ability to search across all messages unless the user account is an admin.

**Comment by atc0005** [comment]
Note to self: appears to be currently tracked by Graylog2/graylog2-server#3473

**Comment by Tafsiralam** [comment]
@reggielamson How and where I have to add this piece of codes..? will you please guide me because I am not able to find the permission settings for the user.

 How can we Implement this .. ?

public static final String SEARCHES_ABSOLUTE = "searches:absolute";
public static final String SEARCHES_KEYWORD = "searches:keyword";
public static final String SEARCHES_RELATIVE = "searches:relative";
public static final String SAVEDSEARCHES_CREATE = "savedsearches:create";
public static final String SAVEDSEARCHES_READ = "savedsearches:read";
public static final String SAVEDSEARCHES_EDIT = "savedsearches:edit";

Thanks in advance

**Comment by cbabs** [comment]
Has this been implemented?  I'm running 2.4.6 and I dont see an option.  I see where this has been merged into milestones, but I see nothing as far as options to configure.  This is a real issue for org wide implementation.

**Comment by Tafsiralam** [comment]
Yes, It's implemented.
